nengo/solvers.py

```diff
@@ -404,8 +408,10 @@ def __call__(self, A, Y, rng=None, E=None):
         import scipy.optimize
 
         tstart = time.time()
-        Y, m, n, d, matrix_in = format_system(A, Y)
-        Y = self.mul_encoders(Y, E)
+        Y, m, n, _, matrix_in = format_system(A, Y)
+        Y = self.mul_encoders(Y, E, copy=True)
+        d = Y.shape[1]
+        Y[Y < 0] = 0  # makes the final RMS error more reasonable
```

Comment:
Oh that's so strange. It does make a difference in the Gram system (if you do it before multiplying Y by the transposed activities), but not in the regular system. I have no idea why that is.